app/views/result/index.html.erb

```diff
@@ -55,105 +55,141 @@
     <%# Sets the base bit of the url and adds the user
         defined search term connected with '+' so it's in
         the right format                                  %>
-    <% url = 'https://www.gov.uk/api/search.json?q=' %>
-    <% term = params["search_term"] %>
-    <% url += term.split.join("+") %>
-
+    <%  url = 'https://www.gov.uk/api/search.json?q='
+        term = params["search_term"]
+        url += term.split.join("+")   %>
 
     <%# Allows the user to not enter a result count (default's 10)
         and caps the displayed results at 1000 as per API limit.
         Also adds the count to the URL that's passed in        %>
-    <% if params["count"] == "" %>
+    <% if params["count"].blank? %>
       <% count = "10" %>
     <% elsif params["count"].to_i > 1000 %>
-      <%count = "1000" %>
+      <% count = "1000" %>
     <% else %>
       <%count = params["count"] %>
     <% end %>
     <% url += "&count=" + count %>
 
 
     <%# Creates an empty hash to put the search results from our 'A'
-        test in and a counter to increment within each of the following
-        loops                           %>
+        test in, also an array to add our 'A' titles and a counter
+        to increment within each of the following loops
+                                  %>
     <% a_list = {} %>
+    <% a_list_titles = [] %>
     <% a_counter = 0 %>
     <% b_counter = 0 %>
-    <div class="ab-search-wrapper-left">
-      <h2 class="results-header">A Results</h2>
-
-      <%# Adds the A test parameter to the URL  %>
-      <% uri = URI("#{url}&ab_tests=#{params["which_test"]}:A") %>
-
-      <%# Gets a result from the search API based on the URL we made %>
-      <% response = Net::HTTP.get(uri) %>
-      <% findings = JSON.parse(response) %>
-
-      <%# Loops over the 'results' section of our result %>
-      <% findings["results"].each do |each_bit| %>
 
-        <%# increments counter and creates hash entry based on link and counter %>
-        <% a_counter += 1 %>
-        <% a_list["#{each_bit["link"]}"] = a_counter %>
+    <%# <h2 class="results-header">A Results</h2> %>
+    <%# Adds the A test parameter to the URL  %>
+    <% uri = URI("#{url}&ab_tests=#{params["which_test"]}:A") %>
+    <%# Gets a result from the search API based on the URL we made %>
+    <% response = Net::HTTP.get(uri) %>
+    <% findings_left = JSON.parse(response) %>
+    <%# Loops over the 'results' section of our result %>
+    <% findings_left["results"].each do |each_bit| %>
+      <%# increments counter and creates hash entry based on link and counter %>
+      <% a_counter += 1 %>
+      <% a_list["#{each_bit["link"]}"] = a_counter %>
+      <% a_list_titles << each_bit["title"] %>
+    <% end %>
+    <% uri = URI("#{url}&ab_tests=#{params["which_test"]}:B") %>
+    <% response = Net::HTTP.get(uri) %>
+    <% findings_right = JSON.parse(response) %>
```

Comment:
Could we extract the above logic into a separate class.

If we do this smart we can get the better objects to use in the looping code below which should help to tidy that up as well.

Something like
```
class ABTestComparer
  def initialize(test_name)
    # code to compare A/B variant results from the server
  end

  def results
    # nicely formatted result (potential) something like the below format
    # might be worth think about what you want the result of the compare to look like in 
    # order to simplify the view code
    [
      [
        a_variant_value,
        position_of_a_variant_in_b_list,
        b_varaint_value,
        posiiton_of_b_varaiant_in_a_list
      ],
      ...
    ]
  end
end